ext/filter/logical_filters.c

```diff
@@ -869,6 +869,106 @@ static int _php_filter_validate_ipv6(const char *str, size_t str_len, int ip[8])
 }
 /* }}} */
 
+/* From the tables in RFC 6890 - Special-Purpose IP Address Registries */
+static bool ipv4_get_status_flags(int ip[8], bool *global, bool *reserved, bool *private)
+{
+	*global = false;
+	*reserved = false;
+	*private = false;
+
+	if (ip[0] == 0) {
+		/* RFC 1122 - This host on this network */
```

Comment:
The errata for RFC6890 changes this, see https://www.rfc-editor.org/errata_search.php?rfc=6890&rec_status=1&presentation=records
Should instead be: `/* RFC 0791 - This network */`.

Reply:
It actually splits up the record into two, so I've made this change instead:

```@@ -869,14 +869,18 @@ static int _php_filter_validate_ipv6(const char *str, size_t str_len, int ip[8])
 }
 /* }}} */
 
-/* From the tables in RFC 6890 - Special-Purpose IP Address Registries */
-static bool ipv4_get_status_flags(int ip[8], bool *global, bool *reserved, bool *private)
+/* From the tables in RFC 6890 - Special-Purpose IP Address Registriesi
+ * Including errata: https://www.rfc-editor.org/errata_search.php?rfc=6890&rec_status=1 */
+static bool ipv4_get_status_flags(const int ip[8], bool *global, bool *reserved, bool *private)
 {
        *global = false;
        *reserved = false;
        *private = false;
 
        if (ip[0] == 0) {
+               /* RFC 0791 - This network */
+               *reserved = true;
+       } else if (ip[0] == 0 && ip[1] == 0 && ip[2] == 0 && ip[3] == 0) {
                /* RFC 1122 - This host on this network */
                *reserved = true;
        } else if (ip[0] == 10) {
```

Reply:
Seems good to me. The compiler will optimize that anyway so it's okay to be verbose :slightly_smiling_face: